pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -586,36 +563,6 @@ type PipelineTaskParam struct {
 	Value string `json:"value"`
 }
 
-// PipelineDeclaredResource is used by a Pipeline to declare the types of the
-// PipelineResources that it will required to run and names which can be used to
-// refer to these PipelineResources in PipelineTaskResourceBindings.
-type PipelineDeclaredResource struct {
-	// Name is the name that will be used by the Pipeline to refer to this resource.
-	// It does not directly correspond to the name of any PipelineResources Task
-	// inputs or outputs, and it does not correspond to the actual names of the
-	// PipelineResources that will be bound in the PipelineRun.
-	Name string `json:"name"`
-	// Type is the type of the PipelineResource.
-	Type PipelineResourceType `json:"type"`
-	// Optional declares the resource as optional.
-	// optional: true - the resource is considered optional
-	// optional: false - the resource is considered required (default/equivalent of not specifying it)
-	Optional bool `json:"optional,omitempty"`
-}
-
-// PipelineTaskResources allows a Pipeline to declare how its DeclaredPipelineResources
-// should be provided to a Task as its inputs and outputs.
-type PipelineTaskResources struct {
-	// Inputs holds the mapping from the PipelineResources declared in
-	// DeclaredPipelineResources to the input PipelineResources required by the Task.
-	// +listType=atomic
-	Inputs []PipelineTaskInputResource `json:"inputs,omitempty"`
-	// Outputs holds the mapping from the PipelineResources declared in
-	// DeclaredPipelineResources to the input PipelineResources required by the Task.
-	// +listType=atomic
-	Outputs []PipelineTaskOutputResource `json:"outputs,omitempty"`
-}
-
 // PipelineTaskInputResource maps the name of a declared PipelineResource input
```

Comment:
Can we not also delete PipelineTaskInputResource and PipelineTaskOutputResource?

Reply:
Sorry I left the wrong struct here to be tombstoned. 

So to confirm that for `PipelineResourecs` do we need to tombstone fields since it has long been deprecated:)

Reply:
don't we need to keep the other types around as well if we want to tombstone?

(Given how long pipeline resouces have been deprecated for, I'm not sure if we need to tombstone)